### PR TITLE
Add Dispatch Task Script

### PR DESCRIPTION
**Pull request by tanyouliang95**
- add `dispatch_task.py` script to submit task to `task_dispatcher_node`
- fix cancel task endpoint and miscellaneous stuffs on `api_server`
- update docs

### CODE REVIEW

**Review by Yadunund — COMMENTED** [review]
I think separating the `dispatch_task` script into three separate scripts: `dispatch_loop`, `dispatch_delivery` and `dispatch_clean` resp. would make their usage more apparent, ie, the supported task capabilities would be clearer. This way we can also get rid of the argument indicating the type of task.

**Comment by tanyouliang95** [comment]
Yeap, done with splitting to `dispatch_loop`, `dispatch_delivery` and `dispatch_clean`

File: rmf_demo_tasks/rmf_demo_tasks/dispatch_clean.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    def generate_task_req_msg(self):
+        req_msg = SubmitTask.Request()
+        req_msg.description.task_type.type = TaskType.TYPE_CLEAN
+
+        clean = Clean()
+        clean.start_waypoint = self.args.clean_start
+        req_msg.description.clean = clean
+
+        ros_start_time = self.node.get_clock().now().to_msg()
```
Comment (Yadunund) [inline rmf_demo_tasks/rmf_demo_tasks/dispatch_clean.py:53]: Refer to comment on `use_sim_time`

**Review by Yadunund — CHANGES_REQUESTED** [review]
Thanks for splitting up the scripts! There is one major problem: The scripts use `self.node.get_clock().now().to_msg()` to populate the `start_time` parameter in the `TaskDescription`. If `use_sim_time` is not enabled in the node that is making the service call, the system time will be used instead. As a result, the task will not execute at the desired time in the simulation. 

As a solution we can parse an argument, `--use_sim_time` which can be defaulted to `True`. Then in `__init__` we can check if this argument is `True` and if so, we call `set_parameters()` on the node

```
from rclpy.parameter import Parameter

        if (self.args.use_sim_time):
            param = Parameter("use_sim_time", Parameter.Type.BOOL, True)
            self.node.set_parameters([param])
```

File: rmf_demo_tasks/rmf_demo_tasks/dispatch_clean.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        try:
+            future = self.submit_task_srv.call_async(req_msg)
+            rclpy.spin_until_future_complete(
+                self.node, future, timeout_sec=1.0)
+            response = future.result()
+            if response is None:
+                self.node.get_logger().error('/submit_task srv call failed')
+            else:
+                self.node.get_logger().info(
```
[the remark this answers is not in the document]
Comment (tanyouliang95) [reply rmf_demo_tasks/rmf_demo_tasks/dispatch_clean.py:87]: Updated! https://github.com/osrf/rmf_demos/pull/199/commits/ddd58e5e04ddf28457a8884c5fa374140dcb6bdb

File: rmf_demo_tasks/rmf_demo_tasks/dispatch_clean.py (hunk 1 above)
Comment (tanyouliang95) [reply rmf_demo_tasks/rmf_demo_tasks/dispatch_clean.py:62]: Thanks for raising this issue that I overlooked it. I have added an optional flag `--disable_sim_time` to those scripts, so the default will to use sim time. Tested and works.

**Review by Yadunund — COMMENTED** [review]
Sorry for being pedantic but can we change the `--disable_sim_time` to `--use_sim_time` instead and flip the logic in the code? Its usage will be more intuitive for ROS users.

**Comment by tanyouliang95** [comment]
The previous intention of using `--disable_sim_time` is to remove the excessive need of having `--use_sim_time` at the end of each script, as they are mainly used in sim. Anyway, according to the prefered choice, I have updated it and the docs accordingly.

**Comment by Yadunund** [comment]
Sorry if I was not clear earlier. I meant we can use `--use_sim_time` as the argument name and set the default value to `True`. That way we do not have to specify the argument at the end of each call.

**Review by Yadunund — APPROVED** [review]
Thanks for making the requested changes! LGTM

**Pull request merged**